kotest-property/src/jvmTest/kotlin/com/sksamuel/kotest/property/arbitrary/DistinctTest.kt

```diff
@@ -1,24 +1,30 @@
 package com.sksamuel.kotest.property.arbitrary
 
 import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.common.DelicateKotest
 import io.kotest.core.spec.style.FunSpec
 import io.kotest.matchers.collections.shouldHaveSize
 import io.kotest.matchers.shouldBe
 import io.kotest.property.Arb
+import io.kotest.property.EdgeConfig
+import io.kotest.property.RandomSource
 import io.kotest.property.arbitrary.arbitrary
+import io.kotest.property.arbitrary.arbitraryBuilder
 import io.kotest.property.arbitrary.distinct
 import io.kotest.property.arbitrary.int
 import io.kotest.property.arbitrary.take
+import kotlin.random.Random
 
+@OptIn(DelicateKotest::class)
 class DistinctTest : FunSpec({
 
    test("with enough entropy distinct should return required count") {
-      Arb.int(0..100000).distinct().take(100).toSet().shouldHaveSize(100)
+      Arb.int(0..100000).distinct().take(100).toList().shouldHaveSize(100)
    }
 
    test("without enough entropy distinct should throw") {
       shouldThrow<NoSuchElementException> {
-         Arb.int(1..9).distinct().take(10).toSet().shouldHaveSize(10)
+         Arb.int(1..9).distinct().take(10).toList().shouldHaveSize(10)
```

Comment:
Why did this change?

Reply:
Because using to set would mask if distinct wasn't working

Reply:
Isn't it the other way around? 

If `distinct().take(2)` returns `[1, 1]` and we do `.toList() shouldHaveSize 2` it feels like distinct is broken but test would still pass.

Reply:
Yeah it needs both, because toSet() will mask duplicates, and toList() will count duplicates.
So I've changed it to use shouldBeUnique() and shouldHaveSize to cover both